[Bug][TubeMQ] Consumer registration failed due to BDB error

### What happened

Consumer registration failed due to BDB error

### What you expected to happen

client log:
2022-11-04 19:40:39.746 - WARN  [           main] .a.i.t.c.c.BaseMessageConsumer:631 [Register Failed] Requested topic [[test_topic_0]] not deployed!
2022-11-04 19:40:39.747 - WARN  [           main] .a.i.t.c.c.BaseMessageConsumer:631 [Register Failed] Requested topic [[test_topic_0]] not deployed!
2022-11-04 19:40:39.748 - WARN  [           main] .a.i.t.c.c.BaseMessageConsumer:631 [Register Failed] Requested topic [[test_topic_0]] not deployed!
2022-11-04 19:40:39.748 -ERROR  [           main] .a.i.t.c.c.BaseMessageConsumer:642 Register to master failed! please check and retry later.
org.apache.inlong.tubemq.client.exception.TubeClientException: Register to master failed! please check and retry later.
        at org.apache.inlong.tubemq.client.consumer.BaseMessageConsumer.startMasterAndBrokerThreads(BaseMessageConsumer.java:644)
        at org.apache.inlong.tubemq.client.consumer.BaseMessageConsumer.completeSubscribe(BaseMessageConsumer.java:301)
        at org.apache.inlong.tubemq.client.consumer.SimplePullMessageConsumer.completeSubscribe(SimplePullMessageConsumer.java:117)
        at org.apache.inlong.tubemq.server.tools.cli.CliConsumer.initTask(CliConsumer.java:261)
        at org.apache.inlong.tubemq.server.tools.cli.CliConsumer.main(CliConsumer.java:384)
2022-11-04 19:40:39.748 -ERROR  [           main] o.a.i.t.s.t.c.CliConsumer     :411 Register to master failed! please check and retry later.

master log:
[INFO] 2022-11-04 19:18:08,042 [DefaultMetaDataService]  [Meta data] trigger broker syncStatus info, brokerId=195060258
[INFO] 2022-11-04 19:18:22,584 [DefBrokerRunManager]  [Broker Release] brokerId=195060258, isTimeout=false, release success!
[INFO] 2022-11-04 19:18:22,584 [DefBrokerRunManager]  [Broker Closed] brokerId=195060258 unregister success, isOverTLS=false
[INFO] 2022-11-04 19:18:36,686 [CmptTMaster]  [Balance Status] 4, isMaster=true, isPrimaryNodeActive=false
[ERROR] 2022-11-04 19:18:46,381 [BdbMetaConfigMapperImpl]  [MetaConfig Error] Connect to node:[test-tube-master-1,TubeMQMasterGroupNode-test-tube-master-1] IOException error is java.net.ConnectException: Connection refused
[ERROR] 2022-11-04 19:18:56,388 [BdbMetaConfigMapperImpl]  [MetaConfig Error] Connect to node:[test-tube-master-1,TubeMQMasterGroupNode-test-tube-master-1] IOException error
java.net.ConnectException: Connection refused
        at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
        at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
        at sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:122)
        at com.sleepycat.je.rep.utilint.RepUtils.openSocketChannel(RepUtils.java:467)
        at com.sleepycat.je.rep.utilint.net.SimpleChannelFactory.connect(SimpleChannelFactory.java:52)
        at com.sleepycat.je.rep.util.DbPing.getNodeState(DbPing.java:309)
        at com.sleepycat.je.rep.util.ReplicationGroupAdmin.getNodeState(ReplicationGroupAdmin.java:430)
        at com.tencent.tubemq.server.master.metamanage.metastore.impl.bdbimpl.BdbMetaConfigMapperImpl.getMasterGroupStatus(BdbMetaConfigMapperImpl.java:334)
        at com.tencent.tubemq.server.master.metamanage.DefaultMetaDataService$2.run(DefaultMetaDataService.java:111)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[ERROR] 2022-11-04 19:19:06,381 [BdbMetaConfigMapperImpl]  [MetaConfig Error] Connect to node:[test-tube-master-1,TubeMQMasterGroupNode-test-tube-master-1] IOException error
java.net.ConnectException: Connection refused
        at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
        at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:716)
        at sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:122)
        at com.sleepycat.je.rep.utilint.RepUtils.openSocketChannel(RepUtils.java:467)
        at com.sleepycat.je.rep.utilint.net.SimpleChannelFactory.connect(SimpleChannelFactory.java:52)
        at com.sleepycat.je.rep.util.DbPing.getNodeState(DbPing.java:309)
        at com.sleepycat.je.rep.util.ReplicationGroupAdmin.getNodeState(ReplicationGroupAdmin.java:430)
        at com.tencent.tubemq.server.master.metamanage.metastore.impl.bdbimpl.BdbMetaConfigMapperImpl.getMasterGroupStatus(BdbMetaConfigMapperImpl.java:334)
        at com.tencent.tubemq.server.master.metamanage.DefaultMetaDataService$2.run(DefaultMetaDataService.java:111)

### How to reproduce

Update to the latest version without clearing metadata

### Environment

_No response_

### InLong version

master

### InLong Component

InLong TubeMQ

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/inlong-tubemq/tubemq-server/src/main/java/org/apache/inlong/tubemq/server/broker/offset/OffsetRecordService.java
+++ b/inlong-tubemq/tubemq-server/src/main/java/org/apache/inlong/tubemq/server/broker/offset/OffsetRecordService.java
@@ -79,7 +79,7 @@ public class OffsetRecordService extends AbstractDaemonService {
         // check topic writable status
         TopicMetadata topicMetadata = storeManager.getMetadataManager()
                 .getTopicMetadata(TServerConstants.OFFSET_HISTORY_NAME);
-        if (!topicMetadata.isAcceptPublish()) {
+        if (topicMetadata == null || !topicMetadata.isAcceptPublish()) {
             return;
         }
         // get group offset information

--- a/inlong-tubemq/tubemq-server/src/main/java/org/apache/inlong/tubemq/server/master/TMaster.java
+++ b/inlong-tubemq/tubemq-server/src/main/java/org/apache/inlong/tubemq/server/master/TMaster.java
@@ -575,6 +575,7 @@ public class TMaster extends HasThread implements MasterService, Stoppable {
             return builder.build();
         }
         final String groupName = (String) paramCheckResult.checkData;
+        checkNodeStatus(consumerId, strBuffer);
         if (!PBParameterUtils.checkConsumerTopicList(defMetaDataService.getDeployedTopicSet(),
                 request.getTopicListList(), result, strBuffer)) {
             builder.setErrCode(result.getErrCode());
@@ -626,7 +627,6 @@ public class TMaster extends HasThread implements MasterService, Stoppable {
             return builder.build();
         }
         ConsumerInfo inConsumerInfo2 = (ConsumerInfo) paramCheckResult.checkData;
-        checkNodeStatus(consumerId, strBuffer);
         CertifiedResult authorizeResult =
                 serverAuthHandler.validConsumerAuthorizeInfo(certResult.userName,
                         groupName, reqTopicSet, reqTopicConditions, rmtAddress);
@@ -1258,6 +1258,8 @@ public class TMaster extends HasThread implements MasterService, Stoppable {
             return builder.build();
         }
         final String groupName = (String) paramCheckResult.checkData;
+        // check master current status
+        checkNodeStatus(consumerId, sBuffer);
         if (!PBParameterUtils.checkConsumerTopicList(defMetaDataService.getDeployedTopicSet(),
                 request.getTopicListList(), result, sBuffer)) {
             builder.setErrCode(result.getErrCode());
@@ -1283,8 +1285,6 @@ public class TMaster extends HasThread implements MasterService, Stoppable {
         if (request.hasOpsTaskInfo()) {
             opsTaskInfo.updOpsSyncInfo(request.getOpsTaskInfo());
         }
-        // check master current status
-        checkNodeStatus(consumerId, sBuffer);
         ClientSyncInfo clientSyncInfo = new ClientSyncInfo();
         if (request.hasSubRepInfo()) {
             clientSyncInfo.updSubRepInfo(brokerRunManager, request.getSubRepInfo());

